oslo_vmware/rw_handles.py

```diff
@@ -558,6 +560,10 @@ def close(self):
         """
         try:
             self._release_lease()
+        except exceptions.ManagedObjectNotFoundException:
+            LOG.info("Lease for %(url)s not found.  No need to release.",
+                     {'url': self._url})
+            return
```

Comment:
You exit here, but there is an extra `close()` call at the end of this function, which would close the http connection.